relationships.md

```diff
@@ -0,0 +1,280 @@
+
+A sample can be found at https://dspace7-entities.atmire.com/rest/#https://dspace7-entities.atmire.com/rest/api/core/relationshiptypes
+
+## Single Relationship
+**/api/core/relationships/<:id>**
+
+A sample can be found at https://dspace7-entities.atmire.com/rest/#https://dspace7-entities.atmire.com/rest/api/core/relationships/530
+The leftId and rightId parameter in JSON may still be present, but they will be removed soon (requires other PRs to be merged first)
+
+```json
+{
+  "id": 530,
+  "relationshipTypeId": 0,
+  "leftPlace": 1,
+  "rightPlace": 1,
+  "type": "relationship",
+  "_links": {
+    "relationshipType": {
+      "href": "https://dspace7-entities.atmire.com/rest/api/core/relationshiptypes/1"
+    },
+    "self": {
+      "href": "https://dspace7-entities.atmire.com/rest/api/core/relationships/530"
+    },
+    "leftItem": {
+      "href": "https://dspace7-entities.atmire.com/rest/api/core/items/e98b0f27-5c19-49a0-960d-eb6ad5287067"
+    },
+    "rightItem": {
+      "href": "https://dspace7-entities.atmire.com/rest/api/core/items/0ffbee3f-e7ea-42bc-92fe-2fbef1a52c0f"
+    }
+  },
+  "_embedded": {
+    "relationshipType": {
+      "id": 1,
+      "leftLabel": "isAuthorOfPublication",
+      "rightLabel": "isPublicationOfAuthor",
+      "leftMinCardinality": 0,
+      "leftMaxCardinality": null,
+      "rightMinCardinality": 0,
+      "rightMaxCardinality": null,
+      "type": "relationshiptype",
+      "_links": {
+        "leftType": {
+          "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/1"
+        },
+        "rightType": {
+          "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/2"
+        },
+        "self": {
+          "href": "https://dspace7-entities.atmire.com/rest/api/core/relationshiptypes/1"
+        }
+      },
+      "_embedded": {
+        "leftType": {
+          "id": 1,
+          "label": "Publication",
+          "type": "entitytype",
+          "_links": {
+            "self": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/1"
+            },
+            "relationshiptypes": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/1/relationshiptypes"
+            }
+          }
+        },
+        "rightType": {
+          "id": 2,
+          "label": "Person",
+          "type": "entitytype",
+          "_links": {
+            "self": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/2"
+            },
+            "relationshiptypes": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/2/relationshiptypes"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+The [relationship type](relationshiptypes.md) is embedded
+The 2 items are included as HAL links but are not embedded
+
+## Relationships per Relationship type
+**/api/core/relationships/search/byLabel?label=<:relationshipname>**
```

Comment:
why repeat label? `/search/byLabel?label=<:relationshipname>`
it will be used in a different way, like an example: `/search/byLabel?id=<:id>`  ? if not, then i think you could dismiss the `/byLabel`

Reply:
Even though there is duplication in `/search/byLabel?label=`, I think we should keep it as-is, since that form aligns better with our REST Contract description of these "/search" sub-endpoints:

https://github.com/DSpace/Rest7Contract/blob/master/search-rels.md

Specifically note that it says:
> All these methods should be exposed under /search/ at /search a HAL document should list all the available search methods for the resources collection

I take this to mean that `/search` must return all the search options available (in this situation, the only option is `byLabel` currently).  That means we need a *subpath* of `/search` to actually perform the search.

So, in summary, I think we should keep this as-is, since I believe changing it to `/search?label=` would cause it to no longer align with current REST Contract documentation.  If we decide to change the REST Contract in the future, we can always go back and modify this endpoint.

Reply:
I understand the need. But if we need to have a subpath of /search and the returned data will be relations or relationships why not change from byLabel to relationships, becoming:

/api/core/relationships/search/relationships?label=<:relationshipname>
or 
/api/core/relationships/search/relations?label=<:relationshipname>

Reply:
@paulo-graca : That again isn't consistent with existing `/api/core/{model}/search` endpoints.  You'd be duplicating *relationships* in that case, as you are already accessing the `/api/core/relationships` endpoint and searching within it for relationships that have a particular field/property.

Here's an existing, documented REST endpoint which acts very similar to this one: `/api/core/metadatafields/search/bySchema?schema=<:prefix>`: https://github.com/DSpace/Rest7Contract/blob/d94482538104ebf826f0cc4c4eecb3a7ce34229b/metadatafields.md#byschema

I'd stress again that I'd like to ensure we are aligning with existing endpoints and not changing currently established "best practices".  If we feel these current "best practices" are wrong, that's a question that can be brought to the DSpace 7 Working Group.  Otherwise, we should attempt to align any Entities work with the practices established by the DSpace 7 Working Group.